futures/src/time.rs

```diff
@@ -62,11 +62,25 @@ where
     ) -> futures::stream::BoxStream<'static, Self::Output> {
         use futures::stream::StreamExt;
 
+        #[cfg(feature = "tokio")]
+        use tokio_new as tokio;
         #[cfg(feature = "tokio_old")]
         use tokio_old as tokio;
 
         let start = tokio::time::Instant::now() + self.0;
 
+        #[cfg(feature = "tokio")]
+        {
+            let stream = async_stream::stream! {
+               let mut interval = tokio::time::interval_at(start, self.0);
+               loop {
+                   yield interval.tick().await;
+               }
+            };
+
+            stream.map(|_| std::time::Instant::now()).boxed()
+        }
```

Comment:
Instead of relying on `async-stream`, could we leverage [`stream::unfold`](https://docs.rs/futures/0.3.8/futures/stream/fn.unfold.html) instead?

Reply:
Alright, did that in daa9dbfccd1fc54010469e9d4d0d06eb02cb262b.